tests/client/test_query.py

```diff
@@ -61,3 +62,198 @@ def test_get_container_sources():
+    etcd_internal_source = SourceLocation(url=etcd_internal_url).save()
+    etcd_build.source_location.connect(etcd_internal_source)
+    etcd_internal_source.upstream.connect(etcd_upstream_source)
+    etcd_component = Component(canonical_type='github', canonical_namespace='coreos',
+                               canonical_name='etcd').save()
+    etcd_upstream_source.component.connect(etcd_component)
+
+    # The container itself
+    container_build = Build(id_='4000', type_='container').save()
+    container_filename = 'docker-image-sha256:' + '1' * 64 + '.x86_64.tar.gz'
+    image = Artifact(
+        archive_id='4001',
+        architecture='x86_64',
+        filename=container_filename,
+        type_='container').save()
+    container_build.artifacts.connect(image)
+    container_internal_url = ('git://pkgs.domain.local/containers/image#...')
+    container_internal_source = SourceLocation(url=container_internal_url).save()
+    container_build.source_location.connect(container_internal_source)
+    image.embedded_artifacts.connect(parent)  # inherits from parent image
+    image.embedded_artifacts.connect(etcd_rpm)  # etcd RPM is installed
+
+    # Go executable
+    # Based on output from backvendor:
+    # *github.com/jimmidyson/configmap-reload@043045da[...] =v0.2.2 ~v0.2.2
+    # golang.org/x/sys@9c60d1c[...] ~v0.0.0-0.20150901164945-9c60d1c508f5
+    # gopkg.in/fsnotify.v1@c282820[...] =v1.4.7 ~v1.4.7
+    configmap_reload = SourceLocation(url='https://github.com/jimmidyson/configmap-reload',
+                                      canonical_version='v0.2.2').save()
+    configmap_reload.component.connect(
+        Component(canonical_type='golang',
+                  canonical_namespace='github.com/jimmidyson',
+                  canonical_name='configmap-reload').save())
+    x_sys = SourceLocation(url='https://go.googlesource.com/sys',
+                           canonical_version='v0.0.0-0.20150901164945-9c60d1c508f5').save()
+    x_sys.component.connect(
+        Component(canonical_type='golang',
+                  canonical_namespace='golang.org/x',
+                  canonical_name='sys').save())
+    fsnotify = SourceLocation(url='https://gopkg.in/fsnotify.v1',
+                              canonical_version='v1.4.7').save()
+    fsnotify.component.connect(
+        Component(canonical_type='golang',
+                  canonical_namespace='gopkg.in',
+                  canonical_name='fsnotify.v1').save())
+
+    container_internal_source.upstream.connect(configmap_reload)
+    container_internal_source.embedded_source_locations.connect(x_sys)
+    container_internal_source.embedded_source_locations.connect(fsnotify)
+
+    expected = {
+        ('container', '4001'): {
+            'artifact': Artifact.nodes.get(archive_id='4001'),
+            'embeds': {
+                ('container', '2001'): {
+                    "artifact": Artifact.nodes.get(archive_id='2001'),
```

Comment:
Why are you returning an `Artifact` object as part of your API function? I thought the API was supposed to return data that could be serialized to JSON.

Reply:
Good point. Maybe the "artifact" key should be replaced by the keys and values of the node's properties? e.g.
```
'embeds': {
  ('container', '2001'): {
    'filename': ...,
    'architecture': ...,
    'sources': ...
    'embeds': ...
  }
}
```

Or is it better to have the properties in their own dict, separated from 'sources' and 'embeds' (which we're adding from other nodes and relationships)?

Reply:
Fixed using option 2.